dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESReadOnlyMonitor.java

```diff
@@ -0,0 +1,192 @@
+            final List<String> usersId = roleAPI.findUsersForRole(adminRole)
+                    .stream()
+                    .map(user -> user.getUserId())
+                    .collect(Collectors.toList());
+
+            final String message = LanguageUtil.get(messageKey);
+
+            final SystemMessageBuilder messageBuilder = new SystemMessageBuilder()
+                    .setMessage(message)
+                    .setSeverity(MessageSeverity.ERROR)
+                    .setType(MessageType.SIMPLE_MESSAGE)
+                    .setLife(TimeUnit.SECONDS.toMillis(5));
+
+            Logger.error(ESReadOnlyMonitor.class, message);
+            systemMessageEventUtil.pushMessage(messageBuilder.create(), usersId);
+        } catch (final LanguageException | DotDataException | DotSecurityException e) {
+            Logger.warn(ESReadOnlyMonitor.class, () -> e.getMessage());
+        }
+    }
+
+    private void putCurrentIndicesToWriteMode() {
+        try {
+            Logger.debug(this.getClass(), () -> "Trying to set the current indices to Write mode");
+            ElasticsearchUtil.setLiveAndWorkingIndicesToWriteMode();
+            sendMessage("es.index.write.allow.message");
+            ReindexThread.setCurrentIndexReadOnly(false);
+
+            this.stop();
+        } catch (final ElasticsearchResponseException e) {
+            Logger.info(ESReadOnlyMonitor.class, ()  -> e.getMessage());
+        }
+    }
+
+    private void putClusterToWriteMode() {
+        Logger.debug(this.getClass(), () -> "Trying to set the current indices to Write mode");
+        ElasticsearchUtil.setClusterToWriteMode();
+        sendMessage("es.cluster.write.allow.message");
+        ReindexThread.setCurrentIndexReadOnly(false);
+
+        this.stop();
+    }
+
+    private void startIndexMonitor() {
+         schedule(new IndexMonitorTimerTask(this));
+    }
+
+    private synchronized void schedule(final TimerTask timerTask) {
+        timer = new Timer(true);
+        timer.schedule(timerTask, 0, TimeUnit.MINUTES.toMillis(1));
+    }
+
+    private void startClusterMonitor() {
+        schedule(new ClusterMonitorTimerTask(this));
+    }
+
+    private synchronized void stop() {
+        if (this.timer != null) {
+            this.timer.cancel();
+            this.timer = null;
+            this.started.set(false);
+        }
+    }
+
+    private static class IndexMonitorTimerTask extends TimerTask{
+        private final ESReadOnlyMonitor esReadOnlyMonitor;
+
+        IndexMonitorTimerTask(final ESReadOnlyMonitor esReadOnlyMonitor) {
+            this.esReadOnlyMonitor = esReadOnlyMonitor;
+        }
+
+        @Override
+        public void run() {
+            this.esReadOnlyMonitor.putCurrentIndicesToWriteMode();
+        }
+    }
+
+    private static class ClusterMonitorTimerTask extends TimerTask{
+        private final ESReadOnlyMonitor esReadOnlyMonitor;
+
+        ClusterMonitorTimerTask(final ESReadOnlyMonitor esReadOnlyMonitor) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [It is a good practice to call super() in a constructor](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5642172)